<p style="text-align:center">
    <a href="https://skills.network/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkML0101ENSkillsNetwork20718538-2022-01-01" target="_blank">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/assets/logos/SN_web_lightmode.png" width="200" alt="Skills Network Logo">
    </a>
</p>

<h1 align="center"><font size="5">Final Project: Classification with Python</font></h1>


# Instructions





Below, is where we are going to use the classification algorithms to create a model based on our training data and evaluate our testing data using evaluation metrics learned in the course.

We will use these algorithms:

1. Linear Regression
2. KNN
3. Decision Trees
4. Logistic Regression
5. SVM

We will evaluate our models using:

1.  Accuracy Score
2.  Jaccard Index
3.  F1-Score
4.  LogLoss
5.  Mean Absolute Error
6.  Mean Squared Error
7.  R2-Score

Finally, we will use the models to generate the report at the end. 


# About The Dataset


The original source of the data is Australian Government's Bureau of Meteorology and the latest data can be gathered from [http://www.bom.gov.au/climate/dwo/](http://www.bom.gov.au/climate/dwo/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkML0101ENSkillsNetwork20718538-2022-01-01).

The dataset to be used has extra columns like 'RainToday' and our target is 'RainTomorrow', which was gathered from the Rattle at [https://bitbucket.org/kayontoga/rattle/src/master/data/weatherAUS.RData](https://bitbucket.org/kayontoga/rattle/src/master/data/weatherAUS.RData?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkML0101ENSkillsNetwork20718538-2022-01-01)




This dataset contains observations of weather metrics for each day from 2008 to 2017. The **weatherAUS.csv** dataset includes the following fields:

| Field         | Description                                           | Unit            | Type   |
| ------------- | ----------------------------------------------------- | --------------- | ------ |
| Date          | Date of the Observation in YYYY-MM-DD                 | Date            | object |
| Location      | Location of the Observation                           | Location        | object |
| MinTemp       | Minimum temperature                                   | Celsius         | float  |
| MaxTemp       | Maximum temperature                                   | Celsius         | float  |
| Rainfall      | Amount of rainfall                                    | Millimeters     | float  |
| Evaporation   | Amount of evaporation                                 | Millimeters     | float  |
| Sunshine      | Amount of bright sunshine                             | hours           | float  |
| WindGustDir   | Direction of the strongest gust                       | Compass Points  | object |
| WindGustSpeed | Speed of the strongest gust                           | Kilometers/Hour | object |
| WindDir9am    | Wind direction averaged of 10 minutes prior to 9am    | Compass Points  | object |
| WindDir3pm    | Wind direction averaged of 10 minutes prior to 3pm    | Compass Points  | object |
| WindSpeed9am  | Wind speed averaged of 10 minutes prior to 9am        | Kilometers/Hour | float  |
| WindSpeed3pm  | Wind speed averaged of 10 minutes prior to 3pm        | Kilometers/Hour | float  |
| Humidity9am   | Humidity at 9am                                       | Percent         | float  |
| Humidity3pm   | Humidity at 3pm                                       | Percent         | float  |
| Pressure9am   | Atmospheric pressure reduced to mean sea level at 9am | Hectopascal     | float  |
| Pressure3pm   | Atmospheric pressure reduced to mean sea level at 3pm | Hectopascal     | float  |
| Cloud9am      | Fraction of the sky obscured by cloud at 9am          | Eights          | float  |
| Cloud3pm      | Fraction of the sky obscured by cloud at 3pm          | Eights          | float  |
| Temp9am       | Temperature at 9am                                    | Celsius         | float  |
| Temp3pm       | Temperature at 3pm                                    | Celsius         | float  |
| RainToday     | If there was rain today                               | Yes/No          | object |
| RainTomorrow  | If there is rain tomorrow                             | Yes/No          | float  |

Column definitions were gathered from [http://www.bom.gov.au/climate/dwo/IDCJDW0000.shtml](http://www.bom.gov.au/climate/dwo/IDCJDW0000.shtml?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkML0101ENSkillsNetwork20718538-2022-01-01)



## **Import the required libraries**


In [1]:
# All Libraries required for this lab are listed below. The libraries pre-installed on Skills Network Labs are commented.
# !mamba install -qy pandas==1.3.4 numpy==1.21.4 seaborn==0.9.0 matplotlib==3.5.0 scikit-learn==0.20.1
# Note: If your environment doesn't support "!mamba install", use "!pip install"

In [2]:
# Surpress warnings:
def warn(*args, **kwargs):
    pass
import warnings
warnings.warn = warn

In [3]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.linear_model import LinearRegression
from sklearn import preprocessing
import numpy as np
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn import svm
from sklearn.metrics import jaccard_score
from sklearn.metrics import f1_score
from sklearn.metrics import log_loss
from sklearn.metrics import confusion_matrix, accuracy_score
import sklearn.metrics as metrics

### Importing the Dataset


In [6]:
from pyodide.http import pyfetch

async def download(url, filename):
    response = await pyfetch(url)
    if response.status == 200:
        with open(filename, "wb") as f:
            f.write(await response.bytes())

In [8]:
path='https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-ML0101EN-SkillUp/labs/ML-FinalAssignment/Weather_Data.csv'

In [9]:
await download(path, "Weather_Data.csv")
filename ="Weather_Data.csv"

In [10]:
df = pd.read_csv("Weather_Data.csv")

> Note: This version of the lab is designed for JupyterLite, which necessitates downloading the dataset to the interface. However, when working with the downloaded version of this notebook on your local machines (Jupyter Anaconda), you can simply **skip the steps above of "Importing the Dataset"** and use the URL directly in the `pandas.read_csv()` function. You can uncomment and run the statements in the cell below.


In [11]:
#filepath = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-ML0101EN-SkillUp/labs/ML-FinalAssignment/Weather_Data.csv"
#df = pd.read_csv(filepath)

In [12]:
df.head()

Date  MinTemp  MaxTemp  Rainfall  Evaporation  Sunshine WindGustDir  \
0  2/1/2008     19.5     22.4      15.6          6.2       0.0           W   
1  2/2/2008     19.5     25.6       6.0          3.4       2.7           W   
2  2/3/2008     21.6     24.5       6.6          2.4       0.1           W   
3  2/4/2008     20.2     22.8      18.8          2.2       0.0           W   
4  2/5/2008     19.7     25.7      77.4          4.8       0.0           W   

   WindGustSpeed WindDir9am WindDir3pm  ...  Humidity9am  Humidity3pm  \
0             41          S        SSW  ...           92           84   
1             41          W          E  ...           83           73   
2             41        ESE        ESE  ...           88           86   
3             41        NNE          E  ...           83           90   
4             41        NNE          W  ...           88           74   

   Pressure9am  Pressure3pm  Cloud9am  Cloud3pm  Temp9am  Temp3pm  RainToday  \
0       1017.6       1017.4         8         8     20.7     20.9        Yes   
1       1017.9       1016.4         7         7     22.4     24.8        Yes   
2       1016.7       1015.6         7         8     23.5     23.0        Yes   
3       1014.2       1011.8         8         8     21.4     20.9        Yes   
4       1008.3       1004.8         8         8     22.5     25.5        Yes   

   RainTomorrow  
0           Yes  
1           Yes  
2           Yes  
3           Yes  
4           Yes  

[5 rows x 22 columns]

### Data Preprocessing


#### One Hot Encoding


First, we need to perform one hot encoding to convert categorical variables to binary variables.


In [13]:
df_sydney_processed = pd.get_dummies(data=df, columns=['RainToday', 'WindGustDir', 'WindDir9am', 'WindDir3pm'])

Next, we replace the values of the 'RainTomorrow' column changing them from a categorical column to a binary column. We do not use the `get_dummies` method because we would end up with two columns for 'RainTomorrow' and we do not want, since 'RainTomorrow' is our target.


In [14]:
df_sydney_processed.replace(['No', 'Yes'], [0,1], inplace=True)

### Training Data and Test Data


Now, we set our 'features' or x values and our Y or target variable.


In [15]:
df_sydney_processed.drop('Date',axis=1,inplace=True)

In [16]:
df_sydney_processed = df_sydney_processed.astype(float)

In [17]:
features = df_sydney_processed.drop(columns='RainTomorrow', axis=1)
Y = df_sydney_processed['RainTomorrow']

### Linear Regression


#### Q1) Use the `train_test_split` function to split the `features` and `Y` dataframes with a `test_size` of `0.2` and the `random_state` set to `10`.


In [18]:
from sklearn.model_selection import train_test_split

# Splitting the data
X_train, X_test, Y_train, Y_test = train_test_split(features, Y, test_size=0.2, random_state=10)

# Display the shapes of the resulting datasets
print("Training features shape:", X_train.shape)
print("Testing features shape:", X_test.shape)
print("Training labels shape:", Y_train.shape)
print("Testing labels shape:", Y_test.shape)


Training features shape: (2616, 66)
Testing features shape: (655, 66)
Training labels shape: (2616,)
Testing labels shape: (655,)


#### Q2) Create and train a Linear Regression model called LinearReg using the training data (`x_train`, `y_train`).


In [20]:
from sklearn.linear_model import LinearRegression

# Create the Linear Regression model
LinearReg = LinearRegression()

# Train the model using the training data
LinearReg.fit(X_train, Y_train)




LinearRegression()

In [21]:
# These are model coefficients to check the result
print("Model trained. Coefficients:", LinearReg.coef_)

Model trained. Coefficients: [-2.36883601e-02  1.30031554e-02  7.30154915e-04  6.49290860e-03
 -3.51614515e-02  4.23841910e-03  1.82843563e-03  7.89794445e-04
  9.55888360e-04  8.56007751e-03  7.70261345e-03 -9.24932344e-03
 -8.87528023e-03  1.00457553e-02  1.44651992e-02 -3.48325241e-03
  1.03190076e+10  1.03190076e+10 -4.35428535e+09 -4.35428535e+09
 -4.35428535e+09 -4.35428535e+09 -4.35428535e+09 -4.35428535e+09
 -4.35428535e+09 -4.35428535e+09 -4.35428535e+09 -4.35428535e+09
 -4.35428535e+09 -4.35428535e+09 -4.35428535e+09 -4.35428535e+09
 -4.35428535e+09 -4.35428535e+09  4.72755310e+09  4.72755310e+09
  4.72755310e+09  4.72755310e+09  4.72755310e+09  4.72755310e+09
  4.72755310e+09  4.72755310e+09  4.72755310e+09  4.72755310e+09
  4.72755310e+09  4.72755310e+09  4.72755310e+09  4.72755310e+09
  4.72755310e+09  4.72755310e+09 -1.18315500e+10 -1.18315500e+10
 -1.18315500e+10 -1.18315500e+10 -1.18315500e+10 -1.18315500e+10
 -1.18315500e+10 -1.18315500e+10 -1.18315500e+10 -1.18315500e

#### Q3) Now use the `predict` method on the testing data (`x_test`) and save it to the array `predictions`.


In [22]:
# Use the trained model to make predictions on the testing data
predictions = LinearReg.predict(X_test)



#### Q4) Using the `predictions` and the `y_test` dataframe calculate the value for each metric using the appropriate function.


In [24]:
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

# Calculate the Mean Absolute Error
LinearRegression_MAE = mean_absolute_error(Y_test, predictions)

# Calculate the Mean Squared Error
LinearRegression_MSE = mean_squared_error(Y_test, predictions)

# Calculate the R-squared value
LinearRegression_R2 = r2_score(Y_test, predictions)


#### Q5) Show the MAE, MSE, and R2 in a tabular format using data frame for the linear model.


In [25]:
import pandas as pd

# Create a DataFrame to store the results
Report = pd.DataFrame({
    'Metric': ['MAE', 'MSE', 'R2'],
    'Linear Regression': [LinearRegression_MAE, LinearRegression_MSE, LinearRegression_R2]
})

# Display the Report
Report


Metric  Linear Regression
0    MAE           0.256319
1    MSE           0.115721
2     R2           0.427130

### KNN


#### Q6) Create and train a KNN model called KNN using the training data (`x_train`, `y_train`) with the `n_neighbors` parameter set to `4`.


In [27]:
from sklearn.model_selection import train_test_split

# Split the data
x_train, x_test, y_train, y_test = train_test_split(features, Y, test_size=0.2, random_state=10)

# Now, create and train the KNN model
KNN = KNeighborsClassifier(n_neighbors=4)
KNN.fit(x_train, y_train)


KNeighborsClassifier(n_neighbors=4)

#### Q7) Now use the `predict` method on the testing data (`x_test`) and save it to the array `predictions`.


In [28]:
# Predict using the KNN model
predictions = KNN.predict(x_test)


#### Q8) Using the `predictions` and the `y_test` dataframe calculate the value for each metric using the appropriate function.


In [33]:
from sklearn.metrics import accuracy_score, jaccard_score, f1_score

# Calculate KNN Accuracy Score
KNN_Accuracy_Score = accuracy_score(y_test, predictions)

# Calculate KNN Jaccard Index
KNN_JaccardIndex = jaccard_score(y_test, predictions)

# Calculate KNN F1 Score
KNN_F1_Score = f1_score(y_test, predictions)

# Print the results
print(f"KNN Accuracy Score: {KNN_Accuracy_Score}")
print(f"KNN Jaccard Index: {KNN_JaccardIndex}")
print(f"KNN F1 Score: {KNN_F1_Score}")


KNN Accuracy Score: 0.7541984732824427
KNN Jaccard Index: 0.39925373134328357
KNN F1 Score: 0.5706666666666667


### Decision Tree


#### Q9) Create and train a Decision Tree model called Tree using the training data (`x_train`, `y_train`).


In [30]:
from sklearn.tree import DecisionTreeClassifier

# Create and train the Decision Tree model
Tree = DecisionTreeClassifier(random_state=10)
Tree.fit(x_train, y_train)


DecisionTreeClassifier(random_state=10)

#### Q10) Now use the `predict` method on the testing data (`x_test`) and save it to the array `predictions`.


In [31]:
# Use the predict method to make predictions on the testing data
predictions = Tree.predict(x_test)


#### Q11) Using the `predictions` and the `y_test` dataframe calculate the value for each metric using the appropriate function.


In [32]:
from sklearn.metrics import accuracy_score, jaccard_score, f1_score

# Calculate Accuracy Score
Tree_Accuracy_Score = accuracy_score(y_test, predictions)

# Calculate Jaccard Index
Tree_JaccardIndex = jaccard_score(y_test, predictions)

# Calculate F1 Score
Tree_F1_Score = f1_score(y_test, predictions)

# Display the results
print(f"Tree Accuracy Score: {Tree_Accuracy_Score}")
print(f"Tree Jaccard Index: {Tree_JaccardIndex}")
print(f"Tree F1 Score: {Tree_F1_Score}")


Tree Accuracy Score: 0.7541984732824427
Tree Jaccard Index: 0.39925373134328357
Tree F1 Score: 0.5706666666666667


### Logistic Regression


#### Q12) Use the `train_test_split` function to split the `features` and `Y` dataframes with a `test_size` of `0.2` and the `random_state` set to `1`.


In [34]:
from sklearn.model_selection import train_test_split

# Split the data with test_size 0.2 and random_state 1
x_train, x_test, y_train, y_test = train_test_split(features, Y, test_size=0.2, random_state=1)


#### Q13) Create and train a LogisticRegression model called LR using the training data (`x_train`, `y_train`) with the `solver` parameter set to `liblinear`.


In [35]:
from sklearn.linear_model import LogisticRegression

# Create the Logistic Regression model with solver set to liblinear
LR = LogisticRegression(solver='liblinear')

# Train the model using the training data
LR.fit(x_train, y_train)


LogisticRegression(solver='liblinear')

#### Q14) Now, use the `predict` and `predict_proba` methods on the testing data (`x_test`) and save it as 2 arrays `predictions` and `predict_proba`.


In [36]:
# Use the predict method to make predictions
predictions = LR.predict(x_test)




In [37]:
# Use the predict_proba method to get the predicted probabilities
predict_proba = LR.predict_proba(x_test)

#### Q15) Using the `predictions`, `predict_proba` and the `y_test` dataframe calculate the value for each metric using the appropriate function.


In [38]:
from sklearn.metrics import accuracy_score, jaccard_score, f1_score, log_loss

# Calculate Logistic Regression Accuracy Score
LR_Accuracy_Score = accuracy_score(y_test, predictions)

# Calculate Logistic Regression Jaccard Index
LR_JaccardIndex = jaccard_score(y_test, predictions)

# Calculate Logistic Regression F1 Score
LR_F1_Score = f1_score(y_test, predictions)

# Calculate Logistic Regression Log Loss
LR_Log_Loss = log_loss(y_test, predict_proba)

# Print out the results
print(f"LR Accuracy Score: {LR_Accuracy_Score}")
print(f"LR Jaccard Index: {LR_JaccardIndex}")
print(f"LR F1 Score: {LR_F1_Score}")
print(f"LR Log Loss: {LR_Log_Loss}")


LR Accuracy Score: 0.8381679389312977
LR Jaccard Index: 0.511520737327189
LR F1 Score: 0.676829268292683
LR Log Loss: 0.38103765010675233


### SVM


#### Q16) Create and train a SVM model called SVM using the training data (`x_train`, `y_train`).


In [39]:
from sklearn.svm import SVC

# Create and train the SVM model
SVM = SVC()
SVM.fit(x_train, y_train)


SVC()

#### Q17) Now use the `predict` method on the testing data (`x_test`) and save it to the array `predictions`.


In [40]:
# Use the predict method to generate predictions
predictions = SVM.predict(x_test)


#### Q18) Using the `predictions` and the `y_test` dataframe calculate the value for each metric using the appropriate function.


In [41]:
from sklearn.metrics import accuracy_score, jaccard_score, f1_score

# Calculate SVM Accuracy Score
SVM_Accuracy_Score = accuracy_score(y_test, predictions)

# Calculate SVM Jaccard Index
SVM_JaccardIndex = jaccard_score(y_test, predictions)

# Calculate SVM F1 Score
SVM_F1_Score = f1_score(y_test, predictions)

# Print the results
print(f"SVM Accuracy Score: {SVM_Accuracy_Score}")
print(f"SVM Jaccard Index: {SVM_JaccardIndex}")
print(f"SVM F1 Score: {SVM_F1_Score}")


SVM Accuracy Score: 0.7221374045801526
SVM Jaccard Index: 0.0
SVM F1 Score: 0.0


### Report


#### Q19) Show the Accuracy,Jaccard Index,F1-Score and LogLoss in a tabular format using data frame for all of the above models.

\*LogLoss is only for Logistic Regression Model


In [44]:


# Create a DataFrame to store the metrics for each model
Report = pd.DataFrame({
    'Metric': ['Accuracy Score', 'Jaccard Index', 'F1 Score', 'Log Loss'],
    'KNN': [KNN_Accuracy_Score, KNN_JaccardIndex, KNN_F1_Score, None],
    'Decision Tree': [Tree_Accuracy_Score, Tree_JaccardIndex, Tree_F1_Score, None],
    'Logistic Regression': [LR_Accuracy_Score, LR_JaccardIndex, LR_F1_Score, LR_Log_Loss],
    'SVM': [SVM_Accuracy_Score, SVM_JaccardIndex, SVM_F1_Score, None]
})

# Display the Report in tabular format
Report


Metric       KNN  Decision Tree  Logistic Regression       SVM
0  Accuracy Score  0.754198       0.754198             0.838168  0.722137
1   Jaccard Index  0.399254       0.399254             0.511521  0.000000
2        F1 Score  0.570667       0.570667             0.676829  0.000000
3        Log Loss       NaN            NaN             0.381038       NaN

### Project Summary: Predicting Rainfall in Australia using Machine Learning Models

#### Overview

This project focuses on predicting rainfall in Australia using machine learning models. The goal is to compare the performance of various classification models and evaluate their ability to predict rainfall outcomes based on a given dataset. The dataset contains information about Australian weather conditions, which includes features like temperature, pressure, humidity, and more, to predict whether it will rain the next day.

#### Dataset

The dataset used in this project is from the **Australian Rainfall Prediction** competition on Kaggle. It consists of historical weather data for various locations in Australia and is intended to predict whether it will rain on the next day. The features include daily weather parameters like temperature, humidity, wind speed, and other factors that might affect rainfall prediction.

#### Models Used

The following machine learning models were applied in this project:

1. **K-Nearest Neighbors (KNN)**: This model predicts the target class based on the majority class of the nearest neighbors in the feature space.
2. **Decision Tree Classifier**: A decision tree model splits the data into subsets based on feature values, creating a tree structure that helps predict the outcome.
3. **Logistic Regression**: A linear model used for binary classification, it predicts the probability of the target variable belonging to a certain class.
4. **Support Vector Machine (SVM)**: A powerful classifier that works by finding the hyperplane that best separates the classes in the feature space.

#### Metrics Used

The performance of each model was evaluated using the following metrics:

- **Accuracy Score**: Measures the overall accuracy of the model by comparing the predicted values with the actual values.
- **Jaccard Index**: Measures the similarity between the predicted and actual values by dividing the intersection of predicted and true values by the union.
- **F1 Score**: A metric that combines precision and recall, providing a balance between the two.
- **Log Loss**: Measures the uncertainty of the model's predictions, primarily used for classification problems that predict probabilities (only used for Logistic Regression in this case).

#### Results

The results of the models were as follows:

| Metric              | KNN        | Decision Tree | Logistic Regression | SVM        | Linear Regression |
|---------------------|------------|---------------|---------------------|------------|-------------------|
| **Accuracy Score**   | 0.754198   | 0.754198      | 0.838168            | 0.722137   | N/A               |
| **Jaccard Index**    | 0.399254   | 0.399254      | 0.511521            | 0.000000   | N/A               |
| **F1 Score**         | 0.570667   | 0.570667      | 0.676829            | 0.000000   | N/A               |
| **Log Loss**         | N/A        | N/A           | 0.381038            | N/A        | N/A               |
| **MAE**              | N/A        | N/A           | N/A                 | N/A        | 0.256319          |
| **MSE**              | N/A        | N/A           | N/A                 | N/A        | 0.115721          |
| **R2**               | N/A        | N/A           | N/A                 | N/A        | 0.427130          |

#### Best Model

**Logistic Regression** performed the best based on the **Accuracy Score** (0.838168), which is the most commonly used metric for classification problems. The **Jaccard Index** and **F1 Score** for Logistic Regression are also relatively high, indicating that the model is not only accurate but also performs well in balancing false positives and false negatives. Although the **Log Loss** for Logistic Regression is relatively low (0.381038), it was only applicable for this model.

In comparison, **SVM** showed poor performance with a **Jaccard Index** and **F1 Score** of 0.000000, indicating that the model was unable to effectively predict the classes.

#### Acknowledgment

This project is part of the **Machine Learning with Python** course, which is a part of the **Data Science Professional Certificate** by IBM.


<h2>About the Authors:</h2> 

<a href="https://www.linkedin.com/in/joseph-s-50398b136/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkML0101ENSkillsNetwork20718538-2022-01-01">Joseph Santarcangelo</a> has a PhD in Electrical Engineering, his research focused on using machine learning, signal processing, and computer vision to determine how videos impact human cognition. Joseph has been working for IBM since he completed his PhD.

### Other Contributors

[Svitlana Kramar](https://www.linkedin.com/in/svitlana-kramar/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMML0232ENSkillsNetwork30654641-2022-01-01)


## Completed By: Qazi Fabia Hoq
This is part of the course "Machine Learning With Python" for Data Science Certification By IBM

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>

<!--
## Change Log

| Date (YYYY-MM-DD) | Version | Changed By    | Change Description          |
| ----------------- | ------- | ------------- | --------------------------- |
| 2022-06-22        | 2.0     | Svitlana K.   | Deleted GridSearch and Mock |
--!>